java/code/src/com/suse/manager/webui/services/impl/SaltService.java

```diff
@@ -1097,6 +1098,27 @@ public Optional<SystemInfo> getSystemInfoFull(String minionId) {
                .map(result -> Json.GSON.fromJson(result, SystemInfo.class));
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PublicCloudInstanceFlavor getInstanceFlavor(String minionId) {
+        LocalCall<String> call = new LocalCall<>("sumautil.instance_flavor",
+                                                 Optional.empty(),
+                                                 Optional.empty(),
+                                                 new TypeToken<>() {  });
+        return callSync(call, minionId)
+            .map(String::toUpperCase)
+            .map(res -> {
+                    if (!res.matches("(PAYG|BYOS|UNKNOWN)")) {
```

Comment:
i think it might be faster to just use `valueOf` here with a try/catch for the error case then matching against a regex and then doing `valueOf`

Reply:
something like this 
```java
.map(res -> {
       try {
            return PublicCloudInstanceFlavor.valueOf(res.toUpperCase());
        } catch (IllegalArgumentException e) {
            return PublicCloudInstanceFlavor.UNKNOWN;
        }
})
```
it can go directly in map and you can remove the first and third map call so it's simpler.